datalad/config.py

```diff
@@ -129,7 +129,7 @@ def _get_default_file_candidates(self):
             cfg_file_candidates.append(opj(home_cfg_base_path, 'datalad.cfg'))
 
         # current dir config
-        cfg_file_candidates.append(opj('.datalad', 'config'))
+        cfg_file_candidates.append('datalad.cfg')
```

Comment:
1. files which shouldn't go under git control should be under `.git/` so they do not pollute git status and would not require custom .gitignore etc
2. so far we also have a number of config settings which should be stored under git:
    - how do we crawl (for now in a custom .datalad/crawl/crawl.cfg)
    - how metadata gets aggregated (somewhere under .datalad/?)

and imho it is not worth placing any datalad files under root dir of a dataset.